Docs(manual): add clarification for default params (#18522)
nim-lang/Nim: doc/manual.rst

  proc foo(a; b: int; c, d: bool): int

A parameter may be declared with a default value which is used if the caller

> does not provide a value for the argument. The value is just replace by the
> compiler and therefore, if applicable, will be reevaluated at call time.

Reviewer: ```suggestion
does not provide a value for the argument. The value will be reevaluated
every time the function is called.
```
"The value is just replace [sic] by the compiler" is very unclear imo.
